gpytorch/lazy/lazy_tensor.py

```diff
@@ -2092,6 +2092,9 @@ def __getitem__(self, index):
         index = tuple(torch.tensor(idx) if isinstance(idx, list) else idx for idx in index)
         index = tuple(idx.item() if torch.is_tensor(idx) and not len(idx.shape) else idx for idx in index)
 
+        # Increase ndimension by the number of dimensions unsqueezed via `None` indexing
+        ndimension += sum(idx is None for idx in index)
+
```

Comment:
I think here it would make sense to call the `self._unsqueeze_batch` for each of the `None` indexes. I think this should be all there is to this PR?

Reply:
Because of https://github.com/cornellius-gp/gpytorch/pull/1813/files that would be infinite recursion. But you are right, in that rather than trying to support `None` indexing in lazy tensors, it might be easier (with smaller chance of breaking things) to implement a non-evaluating `LazyEvaluatedKernelTensor._unsqueeze_batch` directly. (The reason I punted it off to `__getitem__` was because I didn't know `None` indexing wasn't supported.) I tried something basic [1] just on the off chance it worked, it doesn't though. Kind of stuck right now...

[1] 
```python
indexes = (slice(None),) * dim + (None,)
return self.__class__(self.x2, self.x1, kernel=self.kernel[indexes], ...)
```

Reply:
@gpleiss, @wjmaddox any further thoughts here? What's currently on master still breaks the botorch HighOrderGP tests, which I'd like to fix...